Library/Formula/fish.rb

```diff
@@ -15,26 +15,19 @@ class Fish < Formula
     url "https://github.com/fish-shell/fish-shell.git", :shallow => false
 
     depends_on "autoconf" => :build
-    # Indeed, the head build always builds documentation
     depends_on "doxygen" => :build
   end
 
-  skip_clean "share/doc"
-
   def install
     system "autoconf" if build.head?
     # In Homebrew's 'superenv' sed's path will be incompatible, so
     # the correct path is passed into configure here.
     system "./configure", "--prefix=#{prefix}", "SED=/usr/bin/sed"
     system "make", "install"
-  end
 
-  def post_install
-    system "pkill fishd || true"
-  end
-
-  test do
-    system "#{bin}/fish", "-c", "echo"
+    # Generate an inital set of manpage completions.
+    ENV["XDG_DATA_HOME"] = "#{etc}/share/completions"
```

Comment:
I don't think this will do what you want, and also I would recommend against it - may create problems in multiuser environments, for example. fish runs `fish_update_completions` in the background on first launch anyway.